library/datasort.c

```diff
@@ -560,6 +561,80 @@ static int datasort_split_iterator_free(struct eblob_iterate_control *ictl __att
 	return 0;
 }
 
+static int datasort_load_index(struct datasort_cfg *dcfg,
+			       struct eblob_base_ctl *bctl,
+			       struct datasort_chunk *chunk) {
+	// Sanity
+	assert(dcfg);
+	assert(bctl);
+	assert(chunk);
+	assert(chunk->index == NULL); // Assume that index is already empty and we don't need clear previous data
+
+	int err;
+	struct eblob_file_ctl index_ctl = bctl->index_ctl;
+	assert(index_ctl.fd > 0);
+
+	chunk->index = calloc(index_ctl.size, sizeof(struct eblob_disk_control));
+	if (chunk->index == NULL) {
+		return -errno;
+	}
+
+	err = __eblob_read_ll(index_ctl.fd, chunk->index, index_ctl.size, index_ctl.offset);
```

Comment:
Why is `index_ctl.offset` used here? When will it be nonzero?

Reply:
I wanted to write a function for general cases. If you ensure that offset should be always zero I can insert additional assert with that.

Reply:
What is the case when `index_ctl.offset` won't be zero?